File: bindings/java/src/main/java/systems/zlink/service/spot/SpotServiceAttachmentStats.java
/* SPDX-License-Identifier: MPL-2.0 */

package systems.zlink.service.spot;

record SpotServiceAttachmentStats(String channelName, int routerCount,
                                  int pubCount, int subCount,
                                  int autoRouterCount,
                                  int autoPubCount, int autoSubCount) {
}

File: bindings/dotnet/src/Zlink/Runtime/Service/SpotServiceAttachmentStats.cs
// SPDX-License-Identifier: MPL-2.0

using Systems.Zlink.Runtime.Native;

namespace Systems.Zlink;

internal sealed record SpotServiceAttachmentStats(
    string ChannelName,
    uint RouterCount,
    uint PubCount,
    uint SubCount,
    uint AutoRouterCount,
    uint AutoPubCount,
    uint AutoSubCount)
{
    internal static unsafe SpotServiceAttachmentStats FromNative(
        ref ZlinkSpotServiceAttachmentStats native)
    {
        fixed (byte* channelName = native.ChannelName)
        {
            return new SpotServiceAttachmentStats(
                NativeHelpers.ReadFixedString(channelName, 256),
                native.RouterCount, native.PubCount, native.SubCount,
                native.AutoRouterCount, native.AutoPubCount,
                native.AutoSubCount);
        }
    }
}
